### DISCUSSION TITLE
How to update firmware when your BlueRetro doesn't have a microUSB port?

### DISCUSSION THREAD

**Discussion by rykotxet** [darthcloud/BlueRetro#1057] [Q&A]
Hi. A couple of years ago, I bought a BlueRetro for PS1/PS2 to a guy from eBay, so I could use a DS4 out of the box with my PS2. It worked well but I didn't try to update or change configuration of the BlueRetro until recent days. And it really surprised me the huge difficulty to perform a successful bluetooth connection between my BlueRetro and PC/Mobile via blueretro.io! After a lot of tries, I finally was able to update the firmware to 24.04; the firmware upload from PC lasted almost 3 hours, it was just unreliable.

So, taking account how bad are BT connections to PC/Android with my BlueRetro, what other reliable alternative I have to perform a update if my unit doesn't include a microUSB port (see the pic below)? 

![Pic of my BlueRetro PS1/2 opened.](https://i.imgur.com/YoxX1HY.jpg)

**Comment by PSX-MAN** [comment]
I'd try changing to a different BL receiver. And most importantly, make sure you don't have anything else connected to the receiver and only BlueRetro. I once wondered why the update was crashing and taking a long time and then I found out I had wireless headphones connected.  After disconnecting all devices and connecting only the BlueRetro, the transmission was fast and stable.

Maybe this is not your case but I would still try to see if you have cold connections somewhere and possibly revive the tin with solder on the contacts.

That would be a good place to start.

PS: the 24.04 version didn't work very well on the PS1/PS2, it seems to me that the BR receiver's performance has deteriorated slightly. Maybe just a feeling.

**Reply by rykotxet** [reply]
Hi, thanks for the reply. I'm sure I had no active BT connections when I updated the FW. Next time I'll try a CSR dongle I've in the RaspberryPi1B+ instead of the TPLink dongle I've in the PC.

I bought this BlueRetro already mounted, and the soldering job seems well done. BTW, the COM to microUSB adapter should be in the right side of the picture, isn't it? I wonder if it's worth to try to install it myself...

And yes, version 24.04 seems it's working worse than the older version I had before. I noticed ghost inputs and weird random behavior sometimes.

**Comment by majorpbx** [comment]
I ran into this same issue. This may be silly to most but I wanted a power board so I designed the one below. It's the first revision and should have it built soon. Anyway, this again may seem overkill but even the 8BitMod Gamecube and N64 receivers do not have a USB port. So this is my ultimate solution so I don't have to have the systems setup and turned on.

![image](https://github.com/user-attachments/assets/f6532b74-e989-4139-9a67-eab45488539d)
